src/core/DownloadBox.ttslua

```diff
@@ -15,27 +13,20 @@ function onLoad()
   }
 
   -- return to boxes
-  if string.match(notes, "................") == "campaigns/return" then
+  if string.match(self.getName(), "......") == "Return" then
     buttonParameters.position.z = 2
 
-    -- official campaign boxes
-  elseif string.match(notes, ".........") == "campaigns" or self.hasTag("LargeBox") then
+  -- full-sized campaign boxes
+  elseif self.hasTag("LargeBox") then
     buttonParameters.position.z = 6
     buttonParameters.height = 500
     buttonParameters.width = 1700
     buttonParameters.font_size = 350
-
-    -- investigator boxes
-  elseif string.match(notes, ".............") == "investigators" then
-    buttonParameters.position.z = 7
-    buttonParameters.height = 850
-    buttonParameters.width = 3400
-    buttonParameters.font_size = 700
   end
 
   self.createButton(buttonParameters)
 end
 
 function buttonClick_download()
-  Global.call('placeholder_download', { url = self.getGMNotes(), replace = self.guid })
+  Global.call('placeholder_download', { name = self.getName(), url = self.getGMNotes(), replace = self.guid })
```

Comment:
Were you planning to add the URL conversion from the name in this one?  Or even better, the library is keyed by name so there shouldn't be any need to provide a URL at all - just look it up from there based on the name.

Reply:
Updated it to use the library for URLs of community content